src/qtcore/qml/elements/QtQuick/TextEdit.js

```diff
@@ -25,14 +105,135 @@ function QMLTextEdit(meta) {
+        type: 'string',
+        name: 'link'
+    }]);
+    this.linkHovered = Signal([{
+        type: 'string',
+        name: 'link'
+    }]);
+
+    // Methods
+    this.append = function append(text) {
+        this.text += text;
+    };
+
+    this.copy = function copy() {
+        // TODO
+    };
+
+    this.cut = function cut() {
+        this.text =
+            this.text(0, this.selectionStart) + this.text(this.selectionEnd, this.text.length);
+        // TODO
+    };
+
+    this.deselect = function deselect() {
+        //this.selectionStart = -1;
+        //this.selectionEnd = -1;
+        //this.selectedText = null;
+    };
+
+    this.getFormattedText = function getFormattedText(start, end) {
+        this.text = this.text.slice(start, end);
+        // TODO
+        // process text
+        return text;
+    };
+
+    this.getText = function getText(start, end) {
+        return this.text.slice(start, end);
+    };
+
+    this.insert = function getText(position, text) {
+        // TODO
+    };
+
+    this.isRightToLeft = function isRightToLeft(start, end) {
+        // TODO
+    };
+
+    this.linkAt = function linkAt(x, y) {
+        // TODO
+    };
+
+    this.moveCursorSelection = function moveCursorSelection(x, y) {
+        // TODO
+    };
+
+    this.paste = function paste() {
+        // TODO
+    };
+
+    this.positionAt = function positionAt(x, y) {
+        // TODO
+    };
+
+    this.positionToRectangle = function positionToRectangle(position) {
+        // TODO
+    };
+
+    this.redo = function redo() {
+        // TODO
+    };
+
+    this.remove = function remove(start, end) {
+        // TODO
+    };
+
+    this.select = function select(start, end) {
+        // TODO
+    };
+
+    this.selectAll = function selectAll() {
+        // TODO
+    };
+
+    this.selectWord = function selectWord() {
+        // TODO
+    };
+
+    this.undo = function undo() {
+        // TODO
+    };
+
+    var getLineCount = function(self) {
+        return self.text.split(/\n/).length;
+    }
+
+    this.Component.completed.connect(this, function() {
+        this.selectByKeyboard = !this.readOnly;
+        updateValue();
+    });
+
     function updateValue(e) {
         if (self.text != self.dom.firstChild.value) {
             self.text = self.dom.firstChild.value;
         }
+        self.length = self.text.length;
+        self.lineCount = getLineCount(self);
+        updateCss(self);
     }
 
     this.dom.firstChild.oninput = updateValue;
     this.dom.firstChild.onpropertychanged = updateValue;
+
+    this.colorChanged.connect(this, function(newVal) {
+        this.dom.firstChild.style.color = newVal;
+    });
+
+    // The function below is from QMLText
+    this.$drawItem = function(c) {
+        c.save();
+        c.font = fontCss(this.font);
+        c.fillStyle = this.color;
+        c.textAlign = "left";
+        c.textBaseline = "top";
+        c.fillText(this.text, this.left, this.top);
+        c.restore();
+    }
```

Comment:
PR: #175. This will have to be updated after that one gets merged. Or the other way — if this gets merged first, #175 will have to be updated.